openedx_lti_tool_plugin/models.py

```diff
@@ -82,33 +84,84 @@ def __init__(self, *args: tuple, **kwargs: dict):
         self._initial_pii = self.pii
 
     @property
-    def username(self) -> str:
-        """str: Username."""
-        return f'{app_config.name}.{self.uuid}'
+    def short_uuid(self) -> str:
+        """str: Short UUID."""
+        return shortuuid.encode(self.uuid)[:8]
 
     @property
-    def email(self) -> str:
-        """str: Email address."""
-        return f'{self.uuid}@{app_config.name}'
+    def given_name(self) -> str:
+        """str: Given name."""
+        return self.pii.get('given_name', '')
+
+    @property
+    def middle_name(self) -> str:
+        """str: Middle name."""
+        return self.pii.get('middle_name', '')
+
+    @property
+    def family_name(self) -> str:
+        """str: Family name."""
+        return self.pii.get('family_name', '')
 
     @property
     def name(self) -> str:
         """str: Name."""
-        name = self.pii.get('name', '')
-        given_name = self.pii.get('given_name', '')
-        middle_name = self.pii.get('middle_name', '')
-        family_name = self.pii.get('family_name', '')
+        if self.given_name and self.middle_name and self.family_name:
+            return f'{self.given_name} {self.middle_name} {self.family_name}'
+
+        if self.given_name and self.family_name:
+            return f'{self.given_name} {self.family_name}'
+
+        return self.given_name
+
+    @property
+    def username(self) -> str:
+        """str: Username."""
+        if not self.given_name:
+            return f'{self.short_uuid}'
 
-        if name:
-            return name
+        return (
+            f'{self.given_name.lower()[:30]}'
+            f'{self.family_name.lower()[:1]}'
+            f'.{self.short_uuid}'
+        )
 
-        if given_name and middle_name and family_name:
-            return f'{given_name} {middle_name} {family_name}'
+    @property
+    def email(self) -> str:
+        """str: Email address."""
+        return f'{self.uuid}@{app_config.name}'
 
-        if given_name and family_name:
-            return f'{given_name} {family_name}'
+    def user_collision(self) -> bool:
```

Comment:
What do you think of searching if there is already an LTI profile with the same uuid? I would say it's very unlikely for a regular user, that is not associated with an LTI profile, to have that specific username and email.

Reply:
@alexjmpb There is already a unique=True parameter set on the LtiProfile uuid field, and the short uuid is derived from such value so there is no way the uuids are duplicated.

The need for the user_collision method is for the even more unlikely scenario that the username (which contains a short uuid with a bigger collision chance than a regular uuid) already exists, if this verification wasn't added the user could just simply retry the launch. 

I would think making sure there will be no issues is better, and the performance impact will be minimal since this will only be executed when the LtiProfile has no user assigned.

What do you think?, I'm missing something?, do you think there could be another alternative apart from just letting it fail on the rare occurrence of a username collision?